src/pages/core/guides/anchor/how-to-create-a-core-nft-asset.md

```diff
@@ -0,0 +1,454 @@
+
+Constraints add extra validation logic. For example, the `#[account(mut)]` constraint ensures that the `asset` and `payer` accounts are set as mutable, meaning that the data within these accounts can be modified during the instruction.
+
+```rust
+#[derive(Accounts)]
+pub struct CreateAsset<'info> {
+    #[account(mut)]
+    pub asset: Signer<'info>,
+    pub collection: Option<Account<'info, BaseCollectionV1>>,
+    pub authority: Option<Signer<'info>>,
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    /// CHECK: this account will be checked by the mpl_core program
+    pub owner: Option<UncheckedAccount<'info>>,
+    /// CHECK: this account will be checked by the mpl_core program
+    pub update_authority: Option<UncheckedAccount<'info>>,
+    pub system_program: Program<'info, System>,
+    #[account(address = MPL_CORE_ID)]
+    /// CHECK: this account is checked by the address constraint
+    pub mpl_core_program: UncheckedAccount<'info>,
+}
+```
+
+Some accounts in the `CreateAsset` struct are marked as `optional`. This is because, in the definition of the `CreateV2CpiBuilder`, certain accounts can be omitted.
+
+```rust
+/// ### Accounts:
+///
+///   0. `[writable, signer]` asset
+///   1. `[writable, optional]` collection
+///   2. `[signer, optional]` authority
+///   3. `[writable, signer]` payer
+///   4. `[optional]` owner
+///   5. `[optional]` update_authority
+///   6. `[]` system_program
+```
+
+To make the example as flexible as possible, every `optional` account in the program instruction is also treated as `optional` in the `create_core_asset` instruction's account struct.
+
+### Creating the Instruction
+
+The `create_core_asset` function utilizes the inputs from the `CreateAsset` account struct and the `CreateAssetArgs` arg struct that we defined earlier to interact with the `CreateV2CpiBuilder` program instruction.
+
+```rust
+pub fn create_core_asset(ctx: Context<CreateAsset>, args: CreateAssetArgs) -> Result<()> {
+
+  let mpl_core_program = &ctx.accounts.mpl_core_program.to_account_info();
+  let asset = &ctx.accounts.asset.to_account_info();
+  let collection = ctx.accounts.collection.as_ref().map(|tmp| tmp.to_account_info());
+  let authority = ctx.accounts.authority.as_ref().map(|tmp| tmp.to_account_info());
+  let payer = &ctx.accounts.payer.to_account_info();
+  let owner = ctx.accounts.owner.as_ref().map(|tmp| tmp.to_account_info());
+  let update_authority = ctx.accounts.update_authority.as_ref().map(|tmp| tmp.to_account_info());
+  let system_program = &ctx.accounts.system_program.to_account_info();        
+  
+  CreateV2CpiBuilder::new(mpl_core_program)
+    .asset(asset)
+    .collection(collection.as_ref())
+    .authority(authority.as_ref())
+    .payer(payer)
+    .owner(owner.as_ref())
+    .update_authority(update_authority.as_ref())
+    .system_program(system_program)
+    .name(args.name)
+    .uri(args.uri)
+    .invoke()?;
+
+  Ok(())
```

Comment:
This is quite messy and not suitable for a guide. Can you clear this up to make it more in a more readble format people can follow?

If you need the code for this let me know.

Reply:
@L0STE pinging this.

Reply:
Thank you Tony, what do you think is quite messy? The .map(|tmp| ...) stuff? I did it that way because I think it's the most compact way of handling a Some / None .to_account_info response

But if you have a better way to handle that, let me know!